packages/jsActions/nanoflow-actions-native/src/geolocation/RequestLocationPermission.ts

```diff
@@ -14,11 +15,97 @@ import Geolocation, { GeolocationStatic } from "@react-native-community/geolocat
 export async function RequestLocationPermission(): Promise<boolean> {
     // BEGIN USER CODE
 
+    let rnGeolocation: GeolocationServiceStatic | GeolocationStatic;
+
     if (navigator && navigator.product === "ReactNative") {
-        if (!navigator.geolocation) {
-            (navigator.geolocation as GeolocationStatic) = Geolocation;
+        if (NativeModules.RNFusedLocation) {
+            const geolocationService = await import("react-native-geolocation-service");
+            rnGeolocation = geolocationService.default;
+        } else if (NativeModules.RNCGeolocation) {
+            rnGeolocation = Geolocation;
+        } else {
+            return Promise.reject(new Error("Geolocation module couldn't find"));
+        }
+    } else if (navigator && navigator.geolocation) {
+        rnGeolocation = navigator.geolocation;
+    } else {
+        return Promise.reject(new Error("Navigator couldn't find"));
+    }
+
+    const hasPermissionIOS = async (): Promise<boolean> => {
+        const openSetting = (): void => {
+            Linking.openSettings().catch(() => {
+                Alert.alert("Unable to open settings.");
+            });
+        };
+
+        if (!rnGeolocation?.requestAuthorization) {
+            return Promise.reject(new Error("requestAuthorization is undefined"));
+        }
+
+        return rnGeolocation.requestAuthorization("whenInUse").then((status: AuthorizationResult) => {
+            if (status === "granted") {
+                return true;
+            }
+
+            if (status === "denied") {
+                Alert.alert("Location permission denied.");
+            }
+
+            if (status === "disabled") {
+                Alert.alert("Turn on Location Services to allow to determine your location.", "", [
```

Comment:
Language can be improved here. What about `Location Services must be enabled to determine your location.`